youtube_dl/extractor/rte.py

```diff
@@ -60,3 +56,71 @@ def _real_extract(self, url):
             'thumbnail': thumbnail,
             'duration': duration,
         }
+
+
+
+class RteRadioIE(InfoExtractor):
+    IE_NAME = 'rte.ie'
+    IE_DESC = 'Raidió Teilifís Éireann Radio'
+    _VALID_URL = r'''(?x)
+        https?://(?:www\.)?rte\.ie/radio/utils/radioplayer/rteradioweb\.html
+        [#]!rii=(?:[0-9]+)(?:%3A|:)  # Channel ID
```

Comment:
This regex is fairly straightforward and `# Channel ID` comment practically does not bring much benefit since it's not used anywhere. [Here](https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/youtube.py#L248-L277) is the non-trivial one.
There is a cancel build X button on the right hand side https://travis-ci.org/rg3/youtube-dl/builds/100002208 but not sure if it's available for you. Why bother at all?